bioconda_utils/update_pinnings.py

```diff
@@ -1,88 +1,125 @@
-#!/usr/bin/env python
-from conda_build import api
 import re
 import sys
 import os.path
+import logging
+import collections
+import enum
+
+import networkx as nx
+
+from .utils import RepoData, load_conda_build_config, parallel_iter
+
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+
 buildNumberRegex = re.compile("^( +number: +)(\d+)(.*?)$")
 buildNumberJinja = re.compile("^({% +set build +=.+?)(\d+)(.*?%})")
 
-def already_bumped(r, name, version, buildNumber):
-    rsub = r.loc[(r.name == name) &
-             (r.platform != "osx") &
-             (r.version == version)]
```

Comment:
@dpryan79 Using `RepoData().get_package_data()` is actually a lot faster than using `RepoData.df.loc`. The `df` property should really be private, actually. Part of the point of the class was to keep the logic working on the DF in one place, so that if its contents need to change, only one place needs to be considered.